---
title: "Group project LSABD"
output: html_document
date: "2025-11-12"
---

# Introduction

## Background information: Abstract article

Patients with end-stage kidney disease (ESKD) are at high risk of severe COVID-19. Here, we perform longitudinal blood sampling of ESKD haemodialysis patients with COVID-19, collecting samples pre-infection, serially duringinfection, and after clinical recovery. Using plasma proteomics, and RNA-sequencing and ow cytometry of immune cells, we identify transcriptomic and proteomic signatures of COVID-19 severity, and nd distinct temporal molecular pro les in patients with severe disease. Supervised learning reveals that the plasma proteome is a superior indicator of clinical severity than the PBMC transcriptome. We show that a decreasing trajectory of plasma LRRC15 (Leucine-rich repeat containing 15), a proposed co-receptor for SARS-CoV-2, is associated with a more severe clinical course. We observe that two months after the acute infection, patients still display dysregulated gene expression related to vascular, platelet and coagulation pathways, including PF4 (platelet factor 4), which may explain the prolonged thrombotic risk following COVID-19.

## Background information: Methodology 

###Test subjects

All study objects were ESKD patients. 

In the first COVID-19 wave:
  - Group of 53 patients infected with COVID-19 was selected
  - Group of 59 patients not infected with COVID-19 was selected as a negative control
->Serial blood sampling over a period of time

In the second COVID-19 wave:
17 patients in the negative controle groupe got infected with COVID-19
-> Also serial blood sampling.
From these patients we have blood samples from a period the were negative and positive for COVID-19
```{r}
knitr::include_graphics("Distribution patients across groups.png")
```

### Plasma proteomics
The proteome is studied using the SomaScan assay (explained: https://www.youtube.com/watch?v=fg4mlG0nGLw), this technique measures the concentration of a set of target proteins. 
Modified aptamers = Somamers are used -> single stranded nucleic acid molecules with modified bases. These Somameres bind with high affinity to a specific protein, used as an alternative for antibodies, and are used to select a specific set of proteins.
SomaScan v4.1 assay contains 7288 Somameres, 48 Somamers were removed due to QC failure, so the final dataset contains 7240 Somamers representing 6323 unique proteins.

Data Processing Protocol: 48 Somamers were removed due to QC failure, so the final post-QC dataset contained 7240 Somamers representing 6323 unique proteins. Where multiple Somamers related to the same protein, we retained these Somamers for univariate analyses such as differential abundance asnalyses. However, for analyses that considered multiple proteins simultaneously, we selected one Somamer at random to represent each protein. One COVID-19 positive sample in the wave 2 cohort failed QC and was excluded from the analyses. The expression values for each Somamer were inverse-rank normalised prior to downstream analyses. For the Wave 1 cohort, proteomic data were available for 86 samples from 37 COVID-19 positive ESKD patients (median 3 samples per patient, range 1-3), plus 53 non-infected ESKD patients. For the Wave 2 cohort (n = 17 patients), following QC, proteomic data were available for 102 samples collected serially during acute COVID-19 infection (median of 6 samples per patient, range 5–7) and 16 pre-infection samples. For one patient, a pre-infection plasma sample was unavailable. In the original manuscript (https://doi.org/10.1038/s41467-022-35454-4) we performed extensive downstream analysis. These included: i) differential protein abundance analyses comparing COVID-19 cases and controls; ii) differential protein abundance identifying associations with COVID-19 severity; iii) modelling of temporal protein profiles for both protein modules and 232 cytokines; and iv) prediction of COVID-19 severity from the proteome.

COVID positive Wave 1 53 - 16 = 37
COVID Negative Wave 1 59 - 6 = 53
COVID positive wave 2 17 -1

# Part 1: data exploration

```{r}
# import libraries
library(tidyverse)

# Import data sets
combined_dataset <- read.csv("combined_dataset.csv")
features_somameres <- read.csv("feature_meta.csv")

# Heads of both combined_datasets
head(combined_dataset)
head(features_somameres)
```

```{r}
library(tidyverse)
```

```{r}
# Split the combined_dataset in protein expression levels and meta data
combined_dataset_protein <-combined_dataset[,1:7241]
combined_dataset_features <-combined_dataset[,7242:7258]

# Columns: sex, ethnicity, ... converted to factors
combined_dataset_features$sex <- as.factor(combined_dataset_features$sex)
combined_dataset_features$ethnicity <- as.factor(combined_dataset_features$ethnicity)
combined_dataset_features$ihd <- as.factor(combined_dataset_features$ihd)
combined_dataset_features$previous_vte <- as.factor(combined_dataset_features$previous_vte)
combined_dataset_features$copd <- as.factor(combined_dataset_features$copd)
combined_dataset_features$diabetes <- as.factor(combined_dataset_features$diabetes)
combined_dataset_features$smoking <- as.factor(combined_dataset_features$smoking)
combined_dataset_features$cause_eskd <- as.factor(combined_dataset_features$cause_eskd)
combined_dataset_features$WHO_severity <- as.factor(combined_dataset_features$WHO_severity)
combined_dataset_features$WHO_temp_severity <- as.factor(combined_dataset_features$WHO_temp_severity)
combined_dataset_features$fatal_disease <- as.factor(combined_dataset_features$fatal_disease)
combined_dataset_features$case_control <- as.factor(combined_dataset_features$case_control)
combined_dataset_features$radiology_evidence_covid <- as.factor(combined_dataset_features$radiology_evidence_covid)
str(combined_dataset_features)
```

## Subsets data
```{r}
# Make a list with all the negative patients in the first wave
negatives_wave1 <- c()
for (i in 1:length(combined_dataset_features$case_control)){
  if (combined_dataset_features$case_control[i] == "NEGATIVE") {negatives_wave1 <- c(negatives_wave1, combined_dataset_features$individual_id[i])}
}

# Make a list with all the positive patients in the first wave
positives_wave1 <- c()
for (i in 1:length(combined_dataset_features$case_control)){
  id_in_negative_wave1 <- combined_dataset_features$individual_id[i] %in% negatives_wave1
  if (combined_dataset_features$case_control[i] == "POSITIVE" & id_in_negative_wave1 == FALSE) {
    id_in_positive_wave1 <- combined_dataset_features$individual_id[i] %in% positives_wave1
    if (id_in_positive_wave1 == FALSE){
      positives_wave1 <- c(positives_wave1, combined_dataset_features$individual_id[i])
    }
  }
}

# Make a list with all the positive patients in the second wave
positives_wave2 <- c()
for (i in 1:length(combined_dataset_features$case_control)){
  id_in_negative_wave1 <- combined_dataset_features$individual_id[i] %in% negatives_wave1
  if (combined_dataset_features$case_control[i] == "POSITIVE" & id_in_negative_wave1 == "TRUE"){
    id_in_positive_wave2 <- combined_dataset_features$individual_id[i] %in% positives_wave2
    if (id_in_positive_wave2 == FALSE){
      positives_wave2 <- c(positives_wave2, combined_dataset_features$individual_id[i])
    }
  }
}

# Features Wave 1 COVID negative patients
features_wave1_COVID_negative <- combined_dataset_features %>% filter(case_control == "NEGATIVE")

```
To Do:
- missing data uitzoeken
- Somameren uitzoeken
- een manier zoeken om onze dataset op te splitsen in 

```{r}
# Pivot the protein abundances dataframe (combindes_dataset_protein) to a longer dataframe
pivot_longer_proteins <- combined_dataset_protein %>% pivot_longer(!X, names_to = "prots", values_to = "abundances") 

# the range of the abundances (min and max values)
range(pivot_longer_proteins$abundances)
# range: [9.3 ; 428717.2]



```

